app/api/databaseInteraction/insertURL.js

```diff
@@ -3,24 +3,22 @@ var MongoClient = require('mongodb').MongoClient;
 
 // if the original url wasn't in the database, insert it anad return it
 module.exports = function(dbURL, paramUrl) {
-	return new Promise(function(resolve, reject) {
-		MongoClient.connect(dbURL).then(function(db) {
-			// get the sites collection
-			winston.log('info', 'Inserting document...');
-			var collection = db.collection('sites');
-			// insert the url and a newly generated shorturl
-			collection.insert(
-				{'original_url': paramUrl,
-					'short_url': generateShortUrl()
-				})
-					.then(function(result) {
-						db.close();
-						winston.log('info', 'Database closed');
-						resolve(result);
-					});
-		}).catch(function(reason) {
-			reject(reason);
-		});
+	var db;
+	return MongoClient.connect(dbURL).then(function(database) {
+		db = database;
+		winston.log('info', 'Inserting document...');
+		var collection = db.collection('sites');
+		// insert the url and a newly generated shorturl
+		return collection.insert(
+			{
+				'original_url': paramUrl,
+				'short_url': generateShortUrl()
+			}
+		);
+	}).then(function(queryResult) {
+		db.close();
+		winston.log('info', 'Database closed');
+		return queryResult;
```

Comment:
nice! This looks great. However, if this promise before this `then` (the one starting at line 8) fails for some reason, the db will never be closed. You may way to provide a 2nd argument to this `then` that handles previous Promise rejection and still runs db.close. If we had `finally` we could utilize that, but nope.